pkgs/applications/misc/quark-goldleaf/default.nix

```diff
@@ -0,0 +1,97 @@
+{ lib
+, maven
+, fetchFromGitHub
+, copyDesktopItems
+, makeDesktopItem
+, imagemagick
+, wrapGAppsHook
+, jdk
+, gtk3
+}:
+
+let
+  jdk' = jdk.override { enableJavaFX = true; };
+  maven' = maven.override { jdk = jdk'; };
+
+  description = "A GUI tool for transfering files between a computer and a Nintendo Switch running Goldleaf";
+in
+maven'.buildMavenPackage rec {
+  pname = "quark-goldleaf";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "XorTroll";
+    repo = "Goldleaf";
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-gagIQGOiygJ0Onm0SrkbFWaovqWX2WJNx7LpSRheCLM=";
+  };
+
+  sourceRoot = "source/Quark";
+
+  patches = [
+    ./make-deterministic.patch
+    ./remove-pom-jfx.patch
+    ./put-config-in-home.patch
+  ];
+
+  mvnHash = "sha256-gA3HsQZFa2POP9cyJLb1l8t3hrJYzDowhJU+5Xl79p4=";
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    imagemagick # for icon conversion
+    wrapGAppsHook
+  ];
+
+  dontWrapGApps = true; # Don't double wrap the program
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm644 ${./99-quark-goldleaf.rules} $out/etc/udev/rules.d/99-quark-goldleaf.rules
+    install -Dm644 target/Quark.jar $out/share/java/quark-goldleaf.jar
+
+    for size in 16 24 32 48 64 128; do
+      mkdir -p $out/share/icons/hicolor/"$size"x"$size"/apps
+      convert -resize "$size"x"$size" src/main/resources/Icon.png $out/share/icons/hicolor/"$size"x"$size"/apps/quark-goldleaf.png
+    done;
+
+    makeWrapper ${jdk'}/bin/java $out/bin/quark-goldleaf \
+        "''${gappsWrapperArgs[@]}" \
+        --prefix XDG_DATA_DIRS : "${gtk3}/share/gsettings-schemas/${gtk3.name}" \
+        --add-flags "-jar $out/share/java/quark-goldleaf.jar"
+
+    runHook postInstall
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "quark-goldleaf";
+      desktopName = "Quark";
+      comment = description;
+      exec = "quark-goldleaf";
+      icon = "quark-goldleaf";
+      terminal = false;
+      categories = [ "Utility" "Game" "FileTransfer" ];
+      keywords = [ "nintendo" "switch" "goldleaf" ];
+    })
+  ];
+
+  meta = {
+    inherit description;
+    changelog = "https://github.com/XorTroll/Goldleaf/releases/tag/${version}";
+    homepage = "https://github.com/XorTroll/Goldleaf#quark-and-remote-browsing";
+    longDescription = ''
+      ${description}
+
+      For the program to work properly, you will have to install Nintendo Switch udev rules.
+
+      You can either do this by enabling the NixOS module `programs.quark-goldleaf`
+      or by adding the package to `services.udev.packages`
+    '';
+    license = lib.licenses.gpl3Only;
+    mainProgram = "quark-goldleaf";
+    maintainers = with lib.maintainers; [ tomasajt ];
+    platforms = lib.platforms.unix;
```

Comment:
Has this been verified? `unix` is quite large and includes everything. BSDs, Linux, MacOS and Cygwin for Windows. Probably good to look at what platforms `JavaFX` and `udev/rules.d/` are supported on.

Reply:
It works on Darwin, in theory, so I'll do linux ++ darwin